libs/providers/go-feature-flag-web/src/lib/go-feature-flag-web-provider.ts

```diff
@@ -0,0 +1,269 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValue,
+  Logger,
+  OpenFeature,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+} from "@openfeature/web-sdk";
+import {
+  GoFeatureFlagAllFlagRequest,
+  GOFeatureFlagAllFlagsResponse,
+  GoFeatureFlagWebProviderOptions,
+  GOFeatureFlagWebsocketResponse,
+} from "./model";
+import axios from "axios";
+import {transformContext} from "./context-transformer";
+
+export class GoFeatureFlagWebProvider implements Provider {
+  private readonly _websocketPath = "ws/v1/flag/change"
+
+  metadata = {
+    name: GoFeatureFlagWebProvider.name,
+  };
+  events = new OpenFeatureEventEmitter();
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+  // endpoint of your go-feature-flag relay proxy instance
+  private readonly _endpoint: string;
+  // timeout in millisecond before we consider the http request as a failure
+  private readonly _apiTimeout: number;
+
+  // initial delay in millisecond to wait before retrying to connect the websocket
+  private readonly _websocketRetryInitialDelay;
+  // multiplier of _websocketRetryInitialDelay after each failure
+  private readonly _websocketRetryDelayMultiplier;
+  // maximum number of retries
+  private readonly _websocketMaxRetries;
+
+  // _websocket is the reference to the websocket connection
+  private _websocket?: WebSocket;
+  // _flags is the in memory representation of all the flags.
+  private _flags: { [key: string]: ResolutionDetails<FlagValue> } = {};
+
+
+  constructor(options: GoFeatureFlagWebProviderOptions, logger?: Logger) {
+    this._logger = logger;
+    this._apiTimeout = options.apiTimeout || 0; // default is 0 = no timeout
+    this._endpoint = options.endpoint;
+    this._websocketRetryInitialDelay = options.websocketRetryInitialDelay || 100;
+    this._websocketRetryDelayMultiplier = options.websocketRetryDelayMultiplier || 2;
+    this._websocketMaxRetries = options.websocketMaxRetries || 10;
+    // Add API key to the headers
+    if (options.apiKey) {
+      axios.defaults.headers.common['Authorization'] = `Bearer ${options.apiKey}`;
+    }
+  }
+
+  async initialize(context: EvaluationContext): Promise<void> {
```

Comment:
I think you should start your provider in a `NOT_READY` state, and update it once the init is done. This is actually _needed_ in other SDKs, but we have a small bug in the one (IMO) where we initialize every time. This can lead to (in some situations) multiple inits of the same provider.

See [here](https://github.com/open-feature/js-sdk/issues/505).